### PR TITLE
Upgrade geckodriver to 3.0.2 to partially fix CVE-2022-33987

### PR DESCRIPTION
**Pull request by CCongWang**
Signed-off-by: CCongWang <wangcong@umich.edu>

### Description
Upgrade geckodriver to 3.0.2 to use got@11.8.5, which partially fix CVE-2022-33987
 
### Issues Resolved
https://github.com/opensearch-project/OpenSearch-Dashboards/issues/1764
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
    - [X] `yarn test:jest`
    - [X] `yarn test:jest_integration`
    - [x] `yarn test:ftr`
- [ ] New functionality has been documented.
- [X] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by zhongnansu — DISMISSED** [review]
lgtm

**Review by ananzh — APPROVED** [review]
LGTM

File: package.json [hunk 1]
```diff
@@ -370,7 +370,7 @@
     "exit-hook": "^2.2.0",
     "fetch-mock": "^7.3.9",
     "fp-ts": "^2.3.1",
-    "geckodriver": "^3.0.1",
```
Comment (kavilla) [inline package.json:373]: I think we can backport it if we just handle it resolution

**Review by kavilla — COMMENTED** [review]
I think we can keep all the resolutions

**Review by ashwin-pc — APPROVED** [review]

**Review by joshuarrrr — APPROVED** [review]

**Pull request merged**